Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1072 @@
+                    "The history request start date must precede the end date, no history returned"));
+                yield break;
+            }
+
+            if (request.Resolution != Resolution.Minute)
+            {
+                throw new ArgumentException($"SamcoBrokerage.ConvertResolution: Unsupported resolution type: {request.Resolution}");
+            }
+
+            var leanSymbol = request.Symbol;
+            var exchange = _securityProvider.GetSecurity(leanSymbol).Exchange;
+            string symbol = _symbolMapper.GetBrokerageSymbol(leanSymbol);
+            var period = request.Resolution.ToTimeSpan();
+            DateTime latestTime = request.StartTimeUtc;
+
+            do
+            {
+                latestTime = latestTime.AddDays(29);
+                var start = request.StartTimeUtc.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+                var end = latestTime.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+
+                var scrip=_symbolMapper.samcoTradableSymbolList.Where(x => x.Name.ToUpperInvariant() == symbol).First();
+                string endpoint = $"/intraday/candleData?symbolName={symbol}&fromDate={start}&toDate={end}";
+                if (scrip.Instrument == "INDEX")
+                {
+                    endpoint= $"/intraday/indexCandleData?symbolName={symbol}&fromDate={start}&toDate={end}";
+                }
+                var restRequest = new RestRequest(endpoint, Method.GET);
+                var response = _samcoAPI.ExecuteRestRequest(restRequest);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception(
+                        $"SamcoBrokerage.GetHistory: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, " +
+                        $"Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+                }
+
+                // we need to drop the last bar provided by the exchange as its open time is a history request's end time
+                var candles = JsonConvert.DeserializeObject<CandleResponse>(response.Content);
+
+
+                if (candles.intradayCandleData?.Any() == null)
+                {
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "NoHistoricalData",
+                        $"Exchange returned no data for {symbol} on history request " +
+                        $"from {request.StartTimeUtc:s} to {request.EndTimeUtc:s}"));
+                    yield break;
+                }
+
+                foreach (var candle in candles.intradayCandleData)
+                {
+                    if (!exchange.DateTimeIsOpen(candle.dateTime) && !request.IncludeExtendedMarketHours)
+                    {
+                        continue;
+                    }
+                    yield return new TradeBar()
```

Comment:
We are always returning tradeBar data here, what if the history request if of QuoteBar type?